crates/supervisor/core/src/syncnode/error.rs

```diff
@@ -17,8 +17,22 @@ pub enum ManagedNodeError {
     Authentication(#[from] AuthenticationError),
 }
 
+impl PartialEq for ManagedNodeError {
+    fn eq(&self, other: &Self) -> bool {
+        use ManagedNodeError::*;
+        match (self, other) {
+            (Client(a), Client(b)) => format!("{}", a) == format!("{}", b),
```

Comment:
isn't it simpler to just compare `a.to_string() == b.to_string()`? moreover, seems better if `a` and `b` would implement `PartialEq` as well instead of comparing strings

Reply:
should compare error codes here instead. however, can add a todo and fix this as debt since we already discussed replacing error variants wrapping strings as debt issue + match them up with spec rpc errors https://github.com/ethereum-optimism/specs/blob/main/specs/interop/supervisor.md#errors. additionally im wondering if we add a new spec section which describes the same errors as data availability errors without the context of rpc as well, since this seems to feel counterintuitive to think about rpc errors in the context of consensus.